plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShapeSelection.cpp

```diff
@@ -1218,271 +1318,400 @@ bool ProxyShape::doSelect(SelectionUndoHelper& helper, const SdfPathVector& orde
-            helper.m_removedRefs.emplace_back(prim.GetPath(), object);
-        }
+                } else {
+                    // Build a path segment of the USD picked object
+                    Ufe::PathSegment ps_usd(
+                        selectedPath.GetText(), USD_UFE_RUNTIME_ID, USD_UFE_SEPARATOR);
+
+                    // Create a sceneItem
+                    const Ufe::SceneItem::Ptr& si { handler->createItem(proxyShapePath + ps_usd) };
 
-        uint32_t hasNodesToCreate = 0;
-        for (auto prim : insertPrims) {
-            if (prim.IsPseudoRoot()) {
-                // For pseudo root, just modify maya's selection, don't alter
-                // our internal paths
-                newlySelectedPaths.append(MFnDagNode(thisMObject()).fullPathName());
-                addObjToSelectionList(helper.m_newSelection, thisMObject());
-                continue;
+                    // Add the sceneItem to selection
+                    helper.m_newUFESelection.append(si);
+
+                    newlySelectedPaths.append(si->path().string().c_str());
+                }
             }
-            m_selectedPaths.insert(prim.GetPath());
-            MString pathName;
-            MObject object = makeUsdTransformChain_internal(
-                prim,
-                helper.m_modifier1,
-                ProxyShape::kSelection,
-                &helper.m_modifier2,
-                &hasNodesToCreate,
-                &pathName);
-            newlySelectedPaths.append(pathName);
-            addObjToSelectionList(helper.m_newSelection, object);
-            helper.m_insertedRefs.emplace_back(prim.GetPath(), object);
+        } break;
+
+        case MGlobal::kRemoveFromList: {
+            helper.m_newUFESelection = *selection;
+            for (auto& selectedPath : orderedPaths) {
+                // when selecting the root path, select the ProxyShape node using normal maya
+                // selection
+                if (selectedPath == root) {
+                    if (proxyIsSelected)
+                        helper.m_unselectRoot = true;
+                } else {
+                    // Build a path segment of the USD picked object
+                    Ufe::PathSegment ps_usd(
+                        selectedPath.GetText(), USD_UFE_RUNTIME_ID, USD_UFE_SEPARATOR);
+
+                    // Create a sceneItem
+                    const Ufe::SceneItem::Ptr& si { handler->createItem(proxyShapePath + ps_usd) };
+
+                    // Add the sceneItem to selection
+                    helper.m_newUFESelection.remove(si);
+                }
+            }
+        } break;
+
+        case MGlobal::kXORWithList: {
+            helper.m_newUFESelection = *selection;
+            for (auto& selectedPath : orderedPaths) {
+                // when selecting the root path, select the ProxyShape node using normal maya
+                // selection
+                if (selectedPath == root) {
```

Comment:
This difference between using the Maya selection and the UFE selection to deal with the pseudo-root feels tricky.  Why not just add the proxy shape to the UFE selection?  I'm far from being an expert in the AL plugin, so ignore this suggestion if it's not relevant.